### ISSUE TITLE
`pkg-config` not functioning as expected on fresh install and shells

### ISSUE THREAD

**Issue by yonkeltron** [LnL7/nix-darwin#934]
Hello,

When trying to compile various libraries in a fresh nix-darwin install (using the flake approach), I get all manner of missing library errors. This seems to be related to an underlying issue with `pkg-config` as illustrated here, which also shows that the `PKG_CONFIG_PATH` environment variable is not set:

```shell
❯ nix shell 'nixpkgs#libsodium.dev' 'nixpkgs#libsodium' 'nixpkgs#pkg-config'
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish

❯ pkg-config --libs libsodium
Package libsodium was not found in the pkg-config search path.
Perhaps you should add the directory containing `libsodium.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libsodium' found

❯ pkg-config --list-all


❯ echo $PKG_CONFIG_PATH


❯ exit
```

I am on macOS Sonoma 14.4.1 with the following environment:

```shell
❯ nix --version
nix (Nix) 2.18.2

❯ uname -a
Darwin <HOSTNAME> 23.4.0 Darwin Kernel Version 23.4.0: Fri Mar 15 00:10:42 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6000 arm64

```

Please let me know if I can provide any additional information or help with testing.

Love,
+Jonathan

**Comment by Samasaur1** [comment]
Using `nix shell` by itself doesn't run shell hooks. You can either make a dev shell in a flake and use `nix develop`:
```nix
{
  description = "A very basic flake";

  outputs = { self, nixpkgs}:
    let
      allSystems = nixpkgs.lib.systems.flakeExposed;
      forAllSystems = nixpkgs.lib.genAttrs allSystems;
      define = f: forAllSystems (system:
        let
          pkgs = import nixpkgs {
            inherit system;
            config.allowUnfree = true;
          };
        in
          f pkgs
      );
    in
    {
      devShells = define (pkgs: {
        default = pkgs.mkShell {
            buildInputs = [
              pkgs.pkg-config
              pkgs.libsodium
            ];
          };
        }
      );
    };
}
```
or you can use `nix-shell`:
```
nix-shell -p libsodium pkg-config
```

**Comment by yonkeltron** [comment]
Sorry, so does this mean it shouldn't work in my normal, day-to-day shell either? Because that's what i'm struggling with a bit more.

Can confirm it works with `nix-shell -p libsodium pkg-config` but not in my regular shell.

Love,
+Jonathan

**Comment by Samasaur1** [comment]
Yes, installing these packages systemwide will also not run the necessary setup hooks. The "right" way to do per-project dependencies is with a dev shell

**Comment by yonkeltron** [comment]
> The "right" way to do per-project dependencies is with a dev shell

Ok, this mostly makes sense. What about if I want to compile and install a binary for my user? Like even some tool with cargo?

**Comment by Samasaur1** [comment]
> What about if I want to compile and install a binary for my user?

Then you would write a Nix derivation for that package, which would include the dependencies, but you still wouldn't install the dependencies systemwide

**Comment by yonkeltron** [comment]
> > What about if I want to compile and install a binary for my user?
> 
> Then you would write a Nix derivation for that package, which would include the dependencies, but you still wouldn't install the dependencies systemwide

Ok because I have had issues with unavailable libraries trying to install the [bupstash](https://github.com/NixOS/nixpkgs/blob/nixos-23.11/pkgs/tools/backup/bupstash/default.nix) package, which is what first brought this issue to my attention. Did I do something wrong?

**Comment by Samasaur1** [comment]
bupstash works fine for me on aarch64-darwin, but doesn't work on x86_64-darwin. Not sure what's wrong with it there, but if you're on x86_64-darwin, then you didn't do anything wrong

**Comment by yonkeltron** [comment]
Since I am on aarch, it seems something is actually wrong. How can I troubleshoot or provide more info? Seems like this could be a bug after all.

**Comment by Samasaur1** [comment]
What exactly are you doing? I'm running
```
nix shell nixpkgs#bupstash
```
and it's working for me.

Can you let me know what exactly you're running, and provide the output of that command?
